pronto_mia/lib/core/services/deployment_plan_service.dart

```diff
@@ -24,6 +24,23 @@ class DeploymentPlanService {
     return deploymentPlanList;
   }
 
+  Future<DeploymentPlan> getDeploymentPlan(int id) async {
+    final queryVariables = {
+      'id': id,
+    };
+    final data = await (await _graphQLService).query(
+      DeploymentPlanQueries.deploymentPlanById,
+      variables: queryVariables,
+    );
+
+    final dtoList = data['deploymentPlans'] as List<Object>;
+    final deploymentPlanList = dtoList

```

Comment:
Because you only expect 1 entry in the list anyway you could just do `DeploymentPlan.fromJson(dtoList.first)` instead of using map to iterate over a list that has only 1 element.